### DISCUSSION TITLE
Copilot VSCode Extension causes high cpu usage

### DISCUSSION THREAD

**Discussion by lebifeng** [community/community#53846] [Copilot Conversations] [answered]
### Select Topic Area

Bug

### Body

platform: MAC OS Monterey 12.1, M1 chip
VS Code version: 1.77.3
copilot extension version: v1.84.51

when i enabled the copilet extenison in vscode, my cpu usage increases to almost 100%
i tried to reinstall the extenison, it won't help too

**Comment by ncerutti** [comment, marked as answer]
Not sure what the problem might be and there are probably good ways of finding out what's wrong, but I managed to solve something similar by disabling one extension at a time and seeing if the problem persisted. Sometimes extensions do not play nice with the others, so the problem might not necessarily caused by the last one installed.

**Comment by koen-dev** [comment]
I'm having the exact same problem since the last two versions of the GitHub Copilot extension. I've disabled all other extensions and it still happens. It looks like the a process called `rg` is hogging up all my processing power:
<img width="278" alt="image" src="https://user-images.githubusercontent.com/5583604/235431975-a575010c-b8be-4fc1-b398-3d7621414ee1.png">

Affected versions:
- 1.84.51
- 1.83.51

If I install version 1.82.15 this problem goes away.

**Reply by patricktree** [reply]
this helped, thanks!

**Reply by Tete1805** [reply]
Thanks, this helped me too.

**Reply by lebifeng** [reply]
you are right, the rg process keeps scaning my node_modules folder after start, while my project has a huge node_modules folder.

**Comment by Tete1805** [comment]
Same problem here on either WSL or a Kubuntu, downgrading did help, but it's kind of obnoxious for a paid extension.

**Comment by JesusCastroFernandez** [comment]
Es posible que el alto uso de CPU al habilitar la extensión de Copilot en Visual Studio Code se deba a varios factores, como la configuración del hardware de tu computadora, el tamaño del proyecto y la cantidad de archivos que estás trabajando, y el uso de otras extensiones o aplicaciones mientras se usa Copilot.

A continuación, te brindo algunas soluciones que pueden ayudarte a reducir el uso de CPU mientras usas Copilot en Visual Studio Code:

Desactiva otras extensiones de VS Code que no estés utilizando en ese momento. Puedes hacerlo desde la sección de "Extensiones" en el menú de VS Code.

Cierra otras aplicaciones o procesos que no estés utilizando mientras usas VS Code y Copilot. Estos pueden estar consumiendo recursos de la CPU y afectando el rendimiento.

Ajusta la configuración de VS Code para que utilice menos recursos. Puedes hacerlo desde la sección de "Configuración" en el menú de VS Code. Algunas opciones que puedes cambiar son "Editor: Renderizado de línea" y "Editor: Renderizado de caracteres".

Verifica que tengas la versión más reciente de la extensión de Copilot instalada. Si no es así, actualiza la extensión a la versión más reciente.

Si ninguna de estas soluciones resuelve el problema, puedes intentar desinstalar y reinstalar VS Code y Copilot. Asegúrate de hacer una copia de seguridad de tus archivos y configuraciones antes de hacerlo.

Espero que estas soluciones te ayuden a resolver el problema de alto uso de CPU al usar Copilot en VS Code. Si aún tienes problemas, te recomiendo ponerte en contacto con el equipo de soporte de Copilot o VS Code para obtener ayuda adicional.

**Comment by airtonix** [comment]
same problem. 
```sh
> inxi -G -S -C
System:
  Host: brass Kernel: 6.2.9-200.fc37.x86_64 arch: x86_64 bits: 64
    Desktop: GNOME v: 43.4 Distro: Fedora release 37 (Thirty Seven)
CPU:
  Info: 8-core model: AMD Ryzen 7 5800X3D bits: 64 type: MT MCP cache:
    L2: 4 MiB
  Speed (MHz): avg: 2823 min/max: 2200/4549 cores: 1: 2200 2: 2928 3: 2428
    4: 3093 5: 2200 6: 3595 7: 2200 8: 2866 9: 2200 10: 2200 11: 2871 12: 3586
    13: 2200 14: 3597 15: 3593 16: 3426
Graphics:
  Device-1: NVIDIA GA102 [GeForce RTX 3090 Ti] driver: nvidia v: 530.41.03
  Device-2: Logitech StreamCam type: USB
    driver: hid-generic,snd-usb-audio,usbhid,uvcvideo
  Display: x11 server: X.Org v: 1.20.14 with: Xwayland v: 22.1.9 driver: X:
    loaded: nvidia unloaded: fbdev,modesetting,nouveau,vesa
    gpu: nvidia,nvidia-nvswitch resolution: 1: 3440x1440~60Hz
    2: 3440x1440~60Hz
  API: OpenGL v: 4.6.0 NVIDIA 530.41.03 renderer: NVIDIA GeForce RTX 3090
    Ti/PCIe/SSE2
```

enabling copilot causes my cpu and gpu fans to sping up to full.

**Reply by airtonix** [reply]
update: downgrading to `1.82.15` brought my spaceship back to earth.

thanks to @lebifeng 👉🏻 https://github.com/orgs/community/discussions/53846#discussioncomment-5922539

**Comment by LucasArruda** [comment]
The is been going on for a long long time.

I am on 1.86 and still current. But have noticed this since probably when I started using Copilot like months ago (maybe 6?).

My VSCode is basically idle, and just opened, as I was in a meeting. 

And my usage is always 100% or more in exactly one extension (Copilot). And if I terminate, I can see Copilot terminating in vscode and restarting again.

My M1 battery used to last 10 or more hours coding, now lasting a few hours (maybe just 2 or 3) with that 100%+ usage. 

<img width="372" alt="Screenshot 2023-05-16 at 5 04 42 PM" src="https://github.com/community/community/assets/88818/0fe9086f-e0bf-476a-82a4-616a89d6b076">

**Reply by lebifeng** [reply]
downgrade to 1.82.15 actually help

**Reply by rahulchauhan04** [reply]
How did you donwgrade ?

**Reply by rahulchauhan04** [reply]
because now it is happening with my machine too

and i am unable to find this version
<img width="1710" alt="Screenshot 2025-01-06 at 3 19 59 PM" src="https://github.com/user-attachments/assets/23ca3ed7-433e-4ced-8f14-d5249f62f863" />

**Comment by koen-dev** [comment]
Hi everyone 👋 

I wanted to provide an update regarding the high CPU usage issue we've been experiencing with GitHub Copilot on larger projects. Given that this discussion was already marked as resolved, it might not have been catching the attention of those who could assist us.

As such, I've created a new bug report to bring more visibility to this issue. You can find it [here](https://github.com/orgs/community/discussions/56755). I encourage everyone experiencing this issue to contribute their experiences and any potential workarounds they may have found to this new thread.

Let's keep the conversation going there and hopefully get some recognition and a solution from the GitHub team soon.

Thank you for your attention to this matter, and let's continue to work together to improve our experience with GitHub Copilot!

**Comment by nsainaney** [comment]
I ran `Help: Start Extension Bisect` and the culprit for me was github.copilot.chat which is a companion extension. Removing this extension solved the high CPU issue. I'm on an M2 Max and the thing drained my battery (which usually lasts about 14hrs) within 2 hrs!

<img width="466" alt="Screenshot 2024-03-13 at 10 12 46 PM" src="https://github.com/community/community/assets/306293/6d5e9b8b-5582-4e33-97f4-d8621cefe7ae">

**Reply by MichaelMW** [reply]
I'm having the same issue. 
How do you disable "github.copilot.chat" without disabling github.copilot?

**Comment by hsynercn** [comment]
Hi everyone,

I have the same issue, especially when I run an explain command with a selected code snippet I can replicate the issue consistently. This bug is effecting the battery life significantly on Mac Air.

**Comment by indexzet** [comment]
I have the same issue

**Comment by william-uribe-puntoscolombia** [comment]
The same issue:
![image](https://github.com/user-attachments/assets/58dbecb2-3a28-4f3a-9f81-bed88b827ebf)

**Comment by cached** [comment]
I have same issue on MacOS

**Reply by andreic-tache** [reply]
I started having the same issues. Confirmed it via "Help: Start Extension Bisect" command. However, I rarely use it so I decided to uninstall the Copilot Chat extension.

**Comment by tam-nguyen-yum** [comment]
It's AUG 2025 and the issue still persists, heats up my Macbook M4, and it was cooled down right after I disabled the extension
<img width="402" height="220" alt="image" src="https://github.com/user-attachments/assets/6d752579-bcb0-4540-9cfb-b2758fdbef14" />

**Comment by ezeanyicollins** [comment]
Hi all, if you notice this issue about `rg` taking up high processor bandwidth, the best thing to do is to try to end the `rg` processes all at once then check if your chat thread history has grown really long. I have noticed different types of quirks when the chat history grows really long. And I somehow believe this `rg` thing is closely related to Copilot's summarization behavior. If it is the case that you have a really long chat thread then the best thing is to immediately start prepping to take the conversation to a new chat context. That will help with stopping the behavior as I have seen in many cases of my own. So I don't think the best solution is to outright uninstall or disable Copilot since you probably want the goodies it provides ;)
